src/v/resource_mgmt/tests/cpu_profiler_test.cc

```diff
@@ -132,85 +162,82 @@ SEASTAR_TEST_CASE(test_cpu_profiler_enable_override_nested) {
       [](auto acc, auto& s) { return acc + s.occurrences; });
 
     BOOST_REQUIRE_LT(samples_p, samples_10p);
-
-    co_await cp.stop();
 }
 
-SEASTAR_TEST_CASE(test_cpu_profiler_enable_override_abort) {
+SEASTAR_THREAD_TEST_CASE(test_cpu_profiler_enable_override_abort) {
     // Ensure that in cases that the profiler is stopped with on-going overrides
     // that those overrides do not delay the stop.
 
-    ss::sharded<resources::cpu_profiler> cp;
-    co_await cp.start(config::mock_binding(false), config::mock_binding(2ms));
-    co_await cp.invoke_on_all(&resources::cpu_profiler::start);
+    cp_holder holder(false, 2ms);
+    ss::sharded<resources::cpu_profiler>& cp = *holder.cp;
 
     std::chrono::milliseconds wait_ms = 100min;
     auto res_fut = cp.local().collect_results_for_period(wait_ms, std::nullopt);
 
     auto start = ss::steady_clock_type::now();
-    co_await cp.stop();
+    cp.stop().get();
     auto end = ss::steady_clock_type::now();
 
     BOOST_REQUIRE(end - start < 1min);
 }
 
-SEASTAR_TEST_CASE(test_cpu_profiler_enable_override_filter_old_samples) {
+SEASTAR_THREAD_TEST_CASE(test_cpu_profiler_enable_override_filter_old_samples) {
```

Comment:
This test did not stop the service, so it would always crash in when not `NDEBUG` (as a comment in the cmakelists.txt confirms as we disabled it in debug). Not sure how this passed the initial debug unit test run in the original PR.